etcdserver/auth/auth.go

```diff
@@ -448,29 +442,33 @@ func (s *store) DisableAuth() error {
 // is called and returns a new User with these modifications applied. Think of
 // all Users as immutable sets of data. Merge allows you to perform the set
 // operations (desired grants and revokes) atomically
-func (u User) merge(n User) (User, error) {
+func (s *store) merge(ou, nu User) (User, error) {
```

Comment:
can we pass in the password interface? i think there is an interface for this. so we do not need to make this a method on store?